pkg/insights/insights.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insights
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/okteto/okteto/pkg/config"
+	"github.com/okteto/okteto/pkg/k8s/events"
+	"github.com/okteto/okteto/pkg/log/io"
+	"github.com/okteto/okteto/pkg/okteto"
+	corev1 "k8s.io/api/core/v1"
+	eventsv1 "k8s.io/api/events/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// reportingController represents the controller that is reporting the event
+	reportingController = "cli"
+)
+
+// InsightsPublisher is a struct that represents the insights publisher
+type InsightsPublisher struct {
+	k8sClientProvider okteto.K8sClientProvider
+	ioCtrl            io.Controller
+}
+
+// NewInsightsPublisher creates a new insights publisher
+func NewInsightsPublisher(k8sClientProvider okteto.K8sClientProvider, ioCtrl io.Controller) *InsightsPublisher {
+	return &InsightsPublisher{
+		k8sClientProvider: k8sClientProvider,
+		ioCtrl:            ioCtrl,
+	}
+}
+
+// TrackEvent tracks an event in the cluster
+// namespace: the namespace where the event is happening
+// insightType: the type of the event (for example: build, deploy, etc.)
+// data: the data of the event as JSON string
+func (ip *InsightsPublisher) TrackEvent(ctx context.Context, namespace, insightType, data string) {
```

Comment:
It seems that this even should/could be private, right? And the ones to be public could be only the ones for build and deploy